File: app/src/main/java/io/mgba/Presenter/Interfaces/ISettingsPanelPresenter.java
package io.mgba.Presenter.Interfaces;

import android.content.Intent;
import android.os.Bundle;

import permissions.dispatcher.PermissionRequest;

public interface ISettingsPanelPresenter {
    void init(Bundle savedInstanceState);

    void onSaveInstance(Bundle outState);

    void setupFragment();

    void onActivityResult(int requestCode, int resultCode, Intent intent);

    void showFilePicker();

    void showRationaleForStorage(PermissionRequest request);

    String requestPreferencesValue(String key, String defaultValue);

    String getTitle();
}

File: app/src/main/java/io/mgba/presenter/interfaces/ISettingsPanelPresenter.kt
package io.mgba.presenter.interfaces

import android.content.Intent
import android.os.Bundle

import permissions.dispatcher.PermissionRequest

interface ISettingsPanelPresenter {

    val title: String

    fun onSaveInstance(outState: Bundle)

    fun setupFragment()

    fun onActivityResult(requestCode: Int, resultCode: Int, intent: Intent)

    fun showFilePicker()

    fun showRationaleForStorage(request: PermissionRequest)

    fun requestPreferencesValue(key: String, defaultValue: String): String
}
